### PR TITLE
EOS-12739 node management code changes

### PR DESCRIPTION
**Pull request by VijayThakkar132**
Signed-off-by: Vijay Thakkar <vijaykumar.thakkar@seagate.com>

## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-12739
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes/No
  </code>
</pre>
## Problem Description
<pre>
  <code>
    HA: JBOD: Provide Node Management
  </code>
</pre>
## Solution
<pre>
  <code>
    Added node, service and node manager class and raised initial review to confirm approach. Need to do more changes based on the comments.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Unit Testing details here...
  </code>
</pre>

### CODE REVIEW

File: ha/core/cluster/cluster_manager.py [hunk 1]
```diff
@@ -21,7 +21,8 @@
 from cortx.utils.ha.dm.decision_monitor import DecisionMonitor
 
 from ha.core.error import HAUnimplemented
-from ha.core.node.replacement.refresh_context import PcsRefreshContex
```
Comment (ajay-paratmandali) [inline ha/core/cluster/cluster_manager.py:24]: No need of Pcs change in main branch

PcsRefreshContex

avoid it

File: ha/core/node/node_manager.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        self._service_manager = CortxServiceManager()
+
+        node_list = Conf.get(const.HA_GLOBAL_INDEX, "Node.nodelist")
+
+        for node_id in node_list.keys():
+            self.node_instance_list[node_id] = Node(node_id)
+
+        Log.debug(f"node_instance_list {self.node_instance_list}")
+
+        self._decision_monitor = DecisionMonitor()
```
Comment (ajay-paratmandali) [inline ha/core/node/node_manager.py:85]: self._decision_monitor = DecisionMonitor()
self._refresh_context = CortxRefreshContext(self._decision_monitor)

this not part of cortx not needed

File: ha/core/cluster/cluster_manager.py [hunk 3]
```diff
@@ -66,7 +67,7 @@ def __init__(self):
         self._execute = SimpleCommand()
         self._decision_monitor = DecisionMonitor()
         # TODO: add node_manager class to handle query
-        self._refresh_contex = PcsRefreshContex(self._decision_monitor)
```
Comment (ajay-paratmandali) [inline ha/core/cluster/cluster_manager.py:69]: No change needed for class PcsClusterManager(ClusterManager):
in main branch

in main branch CortxClusterManager is used

File: ha/core/node/node_manager.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        Returns instance of the node
+        """
+        if self.validate_node(node_id):
+            node = self.node_instance_list[node_id]
+        else:
+            Log.error(f"Invalid node id {node_id}")
+
+        return node
+
+class PcsNodeManager(NodeManager):
```
Comment (ajay-paratmandali) [inline ha/core/node/node_manager.py:160]: class PcsNodeManager(NodeManager): 
not need in main

File: ha/core/node/node_manager.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        self._refresh_context = CortxRefreshContext(self._decision_monitor)
+
+    def process_request(self, action, args, output):
+        """
+        Generic method to handle process request
+
+        Args:
+            args ([dict]): Parameter pass to request to process.
+        """
+        if action == const.NODE_COMMAND:
```
Comment (ajay-paratmandali) [inline ha/core/node/node_manager.py:95]: const.NODE_COMMAND:
for main node action is raise HAUnimplemented()

**Comment by ajaysrivas** [comment]
Could you please update the description.

File: ha/core/cluster/cluster_manager.py (hunk 1 above)
Comment (VijayThakkar132) [reply ha/core/cluster/cluster_manager.py:24]: Done

File: ha/core/node/node_manager.py (hunk 2 above)
Comment (VijayThakkar132) [reply ha/core/node/node_manager.py]: Done

File: ha/core/cluster/cluster_manager.py (hunk 3 above)
Comment (VijayThakkar132) [reply ha/core/cluster/cluster_manager.py]: Done

File: ha/core/cluster/cluster_manager.py (hunk 1 above)
Comment (VijayThakkar132) [reply ha/core/cluster/cluster_manager.py:24]: Done

File: ha/core/node/node_manager.py (hunk 2 above)
Comment (VijayThakkar132) [reply ha/core/node/node_manager.py]: Done

File: ha/core/cluster/cluster_manager.py (hunk 3 above)
Comment (VijayThakkar132) [reply ha/core/cluster/cluster_manager.py]: Done

File: ha/core/node/node_manager.py (hunk 4 above)
Comment (VijayThakkar132) [reply ha/core/node/node_manager.py]: Removed

File: ha/core/node/node_manager.py (hunk 5 above)
Comment (VijayThakkar132) [reply ha/core/node/node_manager.py:92]: Why it should be unimplemented? Could you please elaborate more?

**Comment by ajaysrivas** [comment]
This is good implementation. This PR must be re-opened or referred to, when we implement the functionality in R2.

**Pull request closed without merging**